docs/local-testing.md

```diff
@@ -4,258 +4,212 @@ In this walkthrough, you will learn how to use the kCTF infrastructure.
 
 In detail, this means:
 
-1. Setting the right `umask`.
-2. Deciding on your test environment ([Docker](#testing-with-docker-only) or [Kubernetes](#testing-with-kubernetes)) and performing the related steps.
-3. [Debugging](#debug-failures) any errors you encountered. 
+1. [First-time setup](#first-time-setup): Setting the right `umask`, installing dependencies, enabling user namespaces.
+2. [Using kCTF](#using-kctf): Downloading the SDK, creating a local cluster and creating tasks.
+3. [Troubleshooting](#Troubleshooting): Basic `kctf chal debug` commands, and `kubectl` usage.
 
-## Set the right umask
+## First-time setup
+
+Following this walkthrough requires a local Linux machine capable of running Docker.
+
+### Set the right umask
 
 Since nsjail will run commands as another user, the challenge files need to be readable by all users. To enable this, set the following umask:
-```
+```bash
 umask a+rx
 ```
 
-## Testing with Docker only
+This should also be setup in our `.bashrc` and `.bash_profile` to ensure you never forget to set it.
 
-Following this walkthrough requires a local Linux machine capable of running Docker.
+```bash
+echo umask a+rx | tee -a ~/.bashrc ~/.bash_profile
+```
 
-This is the fastest way to get started with developing challenges. Alternatively, you can [test with a local Kubernetes cluster](#testing-with-kubernetes) for a more precise emulation of the production environment.
+If you forget to set the right umask, the SDK will warn you.
 
-### Install Docker
+### Install dependencies
+Most people should have `wget`, `curl` and `xxd` installed already, but if you are running on a fresh Debian run:
+```bash
+sudo apt install xxd wget curl netcat
 ```
+
+### Install Docker
+If you have not installed Docker, you should do so by following the [official instructions](https://docs.docker.com/engine/install/).
+
+At the time of writing, one of the supported methods to install docker is by running the following commands:
+```bash
 curl -fsSL https://get.docker.com -o get-docker.sh
 sudo sh get-docker.sh
 sudo usermod -aG docker $USER && newgrp docker
 ```
 
-### Download kCTF
-```
-git clone https://github.com/google/kctf.git
-PATH=$PATH:$PWD/kctf/bin
-```
+If you already have Docker installed, the script will show a warning.
 
-### Create basic demo challenge
-```
-CHALDIR=$(mktemp -d)
-kctf-setup-chal-dir $CHALDIR
-kctf-chal-create test-1
+### Enable user namespaces
+Some Linux distributions don't have user namespaces by default, to enable them in Debian or Ubuntu execute:
+```bash
+echo 'kernel.unprivileged_userns_clone=1' | sudo tee -a /etc/sysctl.d/00-local-userns.conf
```

Comment:
add a note that this has security implications?

Reply:
what are they?

Reply:
increased kernel attack surface, that's why some distros don't enable them by default